experimenter/manifesttool/releases.py

```diff
@@ -64,15 +64,18 @@ def discover_branched_releases(
     app_config: AppConfig,
     strategy: BranchedDiscoveryStrategy,
 ) -> dict[Version, Ref]:
-    assert app_config.repo.type == RepositoryType.HGMO
+    if app_config.repo.type == RepositoryType.GITHUB:
+        resolve_branch = github_api.resolve_branch
+    elif app_config.repo.type == RepositoryType.HGMO:
+        resolve_branch = hgmo_api.resolve_branch
+    else:  # pragma: no cover
+        raise AssertionError()
```

Comment:
Could we add some text in here to make it obvious that it's failing at the repo type?

Reply:
This code is actually unreachable. Its just there to let LSPs know that after this `if-elif-else` branch that `reesolve_branch` is defined.

If repo.type were anything other than HGMO or GITHUB, we would fail in the parsing step in appconfig.py